src/Workleap.OpenApi.MSBuild/SpectralManager.cs

```diff
@@ -0,0 +1,140 @@
+using System.Runtime.InteropServices;
+using Microsoft.Build.Framework;
+using Workleap.OpenApi.MSBuild.Exceptions;
+
+namespace Workleap.OpenApi.MSBuild;
+
+internal sealed class SpectralManager : ISpectralManager
+{
+    private const string SpectralVersion = "6.11.0";
+
+    private readonly IProcessWrapper _processWrapper;
+    private readonly ILoggerWrapper _loggerWrapper;
+    private readonly string _openApiToolsDirectoryPath;
+    private readonly string _toolDirectory;
+
+    public SpectralManager(IProcessWrapper processWrapper, ILoggerWrapper loggerWrapper, string openApiToolsDirectoryPath)
+    {
+        this._processWrapper = processWrapper;
+        this._loggerWrapper = loggerWrapper;
+        this._openApiToolsDirectoryPath = openApiToolsDirectoryPath;
```

Comment:
"this._openApiToolsDirectoryPath" doesn't seems necessary as we already compute this._toolDirectory which includes it.

We do use it here `Directory.CreateDirectory(Path.Combine(this._openApiToolsDirectoryPath, "spectral"));`

But it's already a prefix of `this._toolDirectory`. Also note that Directory.CreateDirectory takes care of creating any missing subdirectory.